ksqldb-rest-app/src/main/java/io/confluent/ksql/api/server/LoggingRateLimiter.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2020 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.api.server;
+
+import static io.confluent.ksql.rest.server.KsqlRestConfig.KSQL_LOGGING_SERVER_RATE_LIMITED_REQUEST_PATHS_CONFIG;
+import static java.util.Objects.requireNonNull;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.util.concurrent.RateLimiter;
+import io.confluent.ksql.rest.server.KsqlRestConfig;
+import io.vertx.ext.web.RoutingContext;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Function;
+
+public class LoggingRateLimiter {
+
+  private final Map<String, Double> rateLimitedPaths;
+  private final Function<Double, RateLimiter> rateLimiterFactory;
+
+  private final Map<String, RateLimiter> rateLimiters = new ConcurrentHashMap<>();
+
+  public LoggingRateLimiter(final KsqlRestConfig ksqlRestConfig) {
+    this(ksqlRestConfig, RateLimiter::create);
+  }
+
+  public LoggingRateLimiter(
+      final KsqlRestConfig ksqlRestConfig,
+      final Function<Double, RateLimiter> rateLimiterFactory) {
+    requireNonNull(ksqlRestConfig);
+    this.rateLimiterFactory = requireNonNull(rateLimiterFactory);
+    this.rateLimitedPaths = getRateLimitedRequestPaths(ksqlRestConfig);
+  }
+
+  public boolean shouldLog(final RoutingContext routingContext) {
+    if (rateLimitedPaths.containsKey(routingContext.request().path())) {
+      final String path = routingContext.request().path();
```

Comment:
Since only the path of the routing context is used here, would it make more sense to pass just the path into this method rather than the entire routing context?